src/data_management/wms_layers/WmsLayerTable.tsx

```diff
@@ -7,116 +7,173 @@ import wmsIcon from "../../images/wms@3x.svg";
+        return new Promise((resolve, _reject) => {
+          resolve();
+        });
+      })
+    });
   }
-}
 
-const deleteAction = (row: any, updateTableRow:any, triggerReloadWithCurrentPage:any, triggerReloadWithBasePage:any)=>{
-  const uuid = row.uuid;
-  if (window.confirm(`Are you sure you want to delete wms-layer with uuid: ${uuid} ?`)) {
-    updateTableRow({...row, markAsDeleted: true});
-    const opts = {
-      credentials: "same-origin",
-      method: "DELETE",
-      headers: { "Content-Type": "application/json" },
-      body: JSON.stringify({})
-    };
-    fetchWmsLayerUuidsWithOptions([uuid], opts)
-    .then((_result) => {
-      // TODO: do we need this callback or should we otherwise indicate that the record is deleted ?
-      triggerReloadWithCurrentPage();
+  const deleteAction = (row: any, updateTableRow:any, triggerReloadWithCurrentPage:any, triggerReloadWithBasePage:any)=>{
+    
+    setRowToBeDeleted(row);
+    setDeleteFunction(()=>()=>{
+      setBusyDeleting(true);
+      updateTableRow({...row, markAsDeleted: true});
+      const opts = {
+        credentials: "same-origin",
+        method: "DELETE",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({})
+      };
+      return fetchWmsLayerUuidsWithOptions([row.uuid], opts)
+      .then((_result) => {
+        setBusyDeleting(false);
+        // TODO: do we need this callback or should we otherwise indicate that the record is deleted ?
+        triggerReloadWithCurrentPage();
+        return new Promise((resolve, _reject) => {
+            resolve();
+          });
+        })
+      
     })
   }
-}
 
-const rasterSourceColumnDefenitions = [
-  {
-    titleRenderFunction: () => "Name",
-    renderFunction: (row: any) => 
-      <span
-        className={tableStyles.CellEllipsis}
-        title={row.name}
-      >
-        <NavLink to={`${navigationUrl}/${row.uuid}/`}>{row.name}</NavLink>
-      </span>,
-    orderingField: "name",
-  },
-  {
-    titleRenderFunction: () =>  "Description",
-    renderFunction: (row: any) => 
-      <span
-        className={tableStyles.CellEllipsis}
-        title={row.description}
-      >
-        {row.description}
-      </span>,
-    orderingField: null,
-  },
-  {
-    titleRenderFunction: () =>  "",//"Actions",
-    renderFunction: (row: any, tableData:any, setTableData:any, triggerReloadWithCurrentPage:any, triggerReloadWithBasePage:any) => {
-      return (
-          <TableActionButtons
-            tableRow={row} 
-            tableData={tableData}
-            setTableData={setTableData} 
-            triggerReloadWithCurrentPage={triggerReloadWithCurrentPage} 
-            triggerReloadWithBasePage={triggerReloadWithBasePage}
-            actions={[
-              {
-                displayValue: "delete",
-                actionFunction: deleteAction,
-              },
-            ]}
-          />
-      );
+  const rasterSourceColumnDefenitions = [
```

Comment:
Also here, the name is incorrect. It should be `wmsLayerColumnDefinitions` instead. Also spelling error here: `Definitions` instead of `Defenitions`. The same in other places.

Reply:
done